### PR TITLE
AoDatePicker component that allows users to type or use pick date picker

### PR DESCRIPTION
**Pull request by sarngru**
# Link to Github Issue
https://github.com/AmpleOrganics/Blaze.vue/issues/312

# Description
AoDatePicker component that allows users to type date or use pick date picker to select a date

# Scope and QA
Following functionality is complete: 
- Date format is: YYYY-MM-DD (2018-01-01)
- User can type in date
- Typed date is synced with date picker
- User can click on icon to toggle date picker
- User can navigate months in date picker using arrows on top left and right
- Styling for selected date and today's date

- [x] Functionality
- [x] Styling
- [x] Unit Tests  

AoDatePicker

### CODE REVIEW

File: src/components/AoDatePicker/AoDatePickerCalendarDay.vue [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+
+  &__header-action {
+    cursor: pointer;
+    padding: .25rem;
+    line-height: 1;
+    border-radius: $border-radius-base;
+    width: $datepicker-arrow-button-width;
+    height: $datepicker-arrow-button-width;
+
+    &:hover {
```
Comment (laurenherdman) [inline src/components/AoDatePicker/AoDatePickerCalendarDay.vue]: when the input is disabled when you hover over the calendar icon the hover styling still applies

Comment (sarngru) [reply src/components/AoDatePicker/AoDatePickerCalendarDay.vue]: @busheikin can you please look into this issue

**Comment by sarngru** [comment]
> is it possible to add a click-outside to close the datePicker? Im thinking of it having an invisible div covering everything and when you click that it emits a close event
> 
> Props like icon-class/html, and is-icon-clickable are they even needed? should they not just be baked in?

@kenzzziez @busheikin can you please comment on this

**Comment by busheikin** [comment]
> > is it possible to add a click-outside to close the datePicker? Im thinking of it having an invisible div covering everything and when you click that it emits a close event
Yes plz

> > Props like icon-class/html, and is-icon-clickable are they even needed? should they not just be baked in?
Makes sense to me but those were added by gru so i'd say wait for his confirmation

**Comment by sarngru** [comment]
> > > is it possible to add a click-outside to close the datePicker? Im thinking of it having an invisible div covering everything and when you click that it emits a close event
> > > Yes plz
> 
> > > Props like icon-class/html, and is-icon-clickable are they even needed? should they not just be baked in?
> > > Makes sense to me but those were added by gru so i'd say wait for his confirmation

Okay, will address these shortly

**Comment by ZainW** [comment]
Screen Shot 2019-03-26 at 11 55 39 AM

Overlap issues when 2 datepickers are with each other

**Comment by sarngru** [comment]
> Screen Shot 2019-03-26 at 11 55 39 AM
> 
> Overlap issues when 2 datepickers are with each other

@busheikin please look into this issue. I have added two date pickers to demo page http://localhost:8081/#/demo to make testing easy.

**Comment by busheikin** [comment]
@sarngru fixed, i think?

**Comment by sarngru** [comment]
> @sarngru fixed, i think?

@busheikin awesome. no z-index issues any more. @ZainW

**Pull request merged**